packages/terra-framework-docs/src/terra-dev-site/doc/worklist-data-grid/Examples.2/RowSelection.jsx

```diff
@@ -78,31 +200,50 @@ const RowSelection = () => {
     return remainingSelectedRow;
   };
 
-  const clearRowSelection = () => {
+  const clearRowSelection = useCallback(() => {
     // eslint-disable-next-line no-param-reassign
     rows.forEach(r => { if (r.isSelected) { r.isSelected = false; } });
     setSelectedRows([]);
-  };
+  }, [rows]);
 
-  const disableSelectableRows = () => {
+  const disableSelectableRows = useCallback(() => {
     rowSelectionModeRef.current.checked = false;
     setHasSelectableRows(false);
     clearRowSelection();
-  };
+  }, [clearRowSelection]);
 
-  const onRowSelectionModeChange = (event) => {
+  const onRowSelectionModeChange = useCallback((event) => {
     if (!event.target.checked) {
       clearRowSelection();
     }
     setHasSelectableRows(event.target.checked);
-  };
+  }, [clearRowSelection]);
 
-  const onColumnSelect = (columnId) => {
+  const onColumnSelect = useCallback((columnId) => {
     if (columnId === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id) {
       // eslint-disable-next-line no-alert
       alert('Row Selection Header Clicked');
     }
-  };
+  }, []);
+
+  const onRowSelect = useCallback((rowIdsToSelect, rowIdsToUnselect) => {
+    rows.forEach((row, index) => {
+      if (rowIdsToUnselect && rowIdsToUnselect.indexOf(row.id) >= 0) {
+        rows[index].isSelected = false;
+      }
+
+      if (rowIdsToSelect && rowIdsToSelect.indexOf(row.id) >= 0) {
+        rows[index].isSelected = true;
+      }
+    });
+  }, [rows]);
```

Comment:
Seems like this logic is a little inefficient.  You have to loop though every row of data even if you only have one row to update and it is the first one.

Reply:
Question: Why are we relying on `indexOf`? Instrinctually if I had a row selection, shouldn't I get this row when I select the row? This could as @cm9361 says also have unintended side effects because we're updating every row, not just the one we select.